api-controllers/ServiceFabrikApiController.js

```diff
@@ -434,21 +434,24 @@ class ServiceFabrikApiController extends FabrikBaseController {
           backup_guid: backupGuid,
           tenant_id: tenantId
         };
-        return timeStamp ?
-          this.backupStore
-          .listBackupsOlderThan(backupFileOptions, new Date(timeStamp))
-          .then(oldBackupArray =>
-            _.sortBy(oldBackupArray, ['started_at']))
-          .then(sortedArray =>
-            _.findLast(sortedArray, backup => backup.state ===  CONST.OPERATION.SUCCEEDED))
-          .tap(successfulBackup => {
-            if (_.isEmpty(successfulBackup)) {
-              throw new NotFound(`No backup successful found for service instance '${instanceId}' before time_stamp ${new Date(timeStamp)} `);
-            }
-          }) :
-          this.backupStore.getBackupFile(backupFileOptions);
+        if (timeStamp) {
+          return this.backupStore
+            .listBackupsOlderThan(backupFileOptions, new Date(timeStamp))
+            .then(oldBackups =>
+              _.sortBy(oldBackups, ['started_at']))
+            .then(sortedOldBackups =>
+              _.findLast(sortedOldBackups, backup => backup.state ===  CONST.OPERATION.SUCCEEDED))
+            .tap(successfulBackup => {
+              if (_.isEmpty(successfulBackup)) {
+                logger.error(`No backup successful found for service instance '${instanceId}' before time_stamp ${new Date(timeStamp)}`);
+                throw new NotFound(`Cannot restore service instance '${instanceId}' as no successful backup found for time_stamp ${timeStamp}`);
```

Comment:
Dont throw errors from `tap` - such errors wont bubble up. Please use tap only for logging, throw errors always from `then` block. I had given similar comments at couple of other places. Please ensure its taken care of.

Reply:
done